### baseline ZERO: "Dumb" reference
- predict mean velocity for every row

In [7]:
import pandas as pd
import numpy as np

In [8]:
df = pd.read_csv("baseline_zero_data.csv")
df.head(20)

,cartridge,bullet_weight,charge_weight,barrel_length,velocity_fps,twist_rate
0,300_blk,200.0,7.0,9.0,710.0,7.0
1,300_blk,200.0,7.5,9.0,877.0,7.0
2,300_blk,200.0,8.5,9.0,1015.0,7.0
3,300_blk,200.0,9.0,9.0,1080.0,7.0
4,300_blk,200.0,11.0,9.0,904.0,7.0
5,300_blk,200.0,11.5,9.0,945.0,7.0
6,300_blk,200.0,11.5,9.0,925.0,7.0
7,300_blk,200.0,12.0,9.0,905.0,7.0
8,300_blk,200.0,11.0,9.0,823.0,7.0
9,300_blk,200.0,11.5,9.0,890.0,7.0


In [10]:
target = df["velocity_fps"].astype(float)  # define target and cast as float

In [18]:
mean_velocity = target.mean()  # this is our dumb predictor, mean value only across all rows

In [19]:
mean_velocity

np.float64(2031.1733333333334)

In [17]:
baseline_zero_predictions = np.full_like(target, mean_velocity)
baseline_zero_mae = np.mean(np.abs(target - baseline_zero_predictions))
baseline_zero_mae

np.float64(774.2243555555557)

### baseline zero mae = 774.224 = absolute floor
- This is the worst the model should behave as it has zero features to work with. If future models yield worse errors then something in them is broken.